### DISCUSSION TITLE
Reuse endpoints using YAML configurations

### DISCUSSION THREAD

**Discussion by cdia8890** [apache/camel-karavan#1264] [Q&A]
### Ask a question

Is there a possibility to reuse endpoint configurations through YAML definitions across routes?

For example, if there is a database/Kafka cluster configuration that needs to be re-used in multiple routes, is there a way to define the endpoint configuration once, and refer the same endpoint configuration in multiple YAML files?

### Variant

Vs Code Extension

### Container Management (if applicable)

Docker

### Operating System (if applicable)

Windows

### Version

4.5.0

**Comment by mgubaidullin** [comment]
1.  you could use Datasource bean 
2. you could create custom Kamelet

**Reply by davsclaus** [reply]
Karavan is just using Camel, and this is not possible in Camel.

**Reply by cdia8890** [reply]
In Camel, I created parameterized functions which return String URI as the output. Then I used those functions in the routes to get the URI definitions dynamically.

Anyway in Karavan I created separate routes for endpoints and am using `direct` to call those routes from other routes. That way I can get some reusability rather than defining entire flow in a single route.
